apps/cyberstorm-nextjs/utils/auth.ts

```diff
@@ -0,0 +1,145 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+import hkdf from "@panva/hkdf";
+import { EncryptJWT, jwtDecrypt } from "jose";
+import { v4 as uuid } from "uuid";
+
+interface UserStatusType {
+  username: string;
+  capabilities: string[];
+  subscription: { expires: string | null };
+  rated_packages: string[];
+  teams: string[];
+}
+
+function isUserStatusType(data: any): boolean {
```

Comment:
- Nitpick: not a fan of `data` as a variable name even in this context
- Type `unknown` would be nice here since the whole point of the function is that we don't know the type of the argument and want to find out by duck typing it. If I'm not mistaken, changing the type would also reveal that this function breaks if the argument is undefined
- The whole point of a type guard is to give TS compiler more information about the typing. Now the function returns boolean, which tells TS squat. Consequently you need to manually type cast the value even after this function is called. Check the documentation I linked to see how this should actually be done 😉
- The type checking in the function body could be improved by e.g. using Array.isArray

Reply:
This part of the code has changed few times because of discussions, but anyways I've fixed the stuff you mentioned there 👌 And thanks for the more in depth discussion on discord also 💯